tests/functions/test_base.py

```diff
@@ -1,10 +1,42 @@
 import pytest
+import mock
+
+import logging
 
 from exodus_lambda.functions.base import LambdaBase
+from test_utils.test_utils import generate_test_config
 
-CONF_PATH = "exodus_lambda/functions/lambda_config.json"
+CONF_PATH = "configuration/lambda_config.json"
+MOCKED_CONF = generate_test_config(CONF_PATH)
 
 
 def test_base():
     with pytest.raises(NotImplementedError):
         LambdaBase(conf_file=CONF_PATH).handler(event=None, context=None)
+
+
+@mock.patch("exodus_lambda.functions.base.json.loads")
```

Comment:
Could we please avoid this mocking?  This bakes a dependency into many tests that json.loads is used for loading config files (and not for anything else), despite that those tests should have no interest in exactly how the config is loaded.

If the idea is that it's too annoying to write the config dict to a file for load: we control the class under test, so can we just make it accept a dict in the first place?  "A class should be testable using only its public API" is a desirable goal, it's valid to tweak the API a little to serve this purpose.  If the class is cumbersome to test and requires mocking just to initialize it with desired config, then we didn't design it well for our own purposes.

My suggestion is that conf_file should be handled as:

- if it's a dict, then it's the config
- else it's a path to a JSON file